lms/static/js/views/fields.js

```diff
@@ -495,6 +495,203 @@
+                    data.submit();
+                }
+            },
+
+            validateImageSize: function (imageBytes) {
+                if (imageBytes < this.options.imageMinBytes) {
+                    this.showUserErrorMessage(gettext("Minimum file size not met."));
+                    return false;
+                } else if (imageBytes > this.options.imageMaxBytes) {
+                    this.showUserErrorMessage(gettext("Maximum file size exceeded."));
+                    return false;
+                }
+                return true;
+            },
+
+            showUploadInProgressMessage: function () {
+                this.$('.upload-button-wrapper').css('opacity', 1);
+                this.$('.upload-button-icon').html(this.iconProgress);
+                this.$('.upload-button-title').html(gettext("Uploading..."));
+            },
+
+            showRemovalInProgressMessage: function () {
+                this.$('.u-field-remove-button').css('opacity', 1);
+                this.$('.remove-button-icon').html(this.iconProgress);
+                this.$('.remove-button-title').html(gettext("Removing..."));
+            },
+
+            setCurrentStatus: function (status) {
+                this.$('.image-wrapper').attr('data-status', status);
+            },
+
+            getCurrentStatus: function () {
+                return this.$('.image-wrapper').attr('data-status');
+            },
+
+            showUserErrorMessage: function (message) {
+                 var messageBannerView = new MessageBannerView({
+                    el: '.error-message-banner',
+                    message: message
+                });
+                messageBannerView.render();
+            },
+
+            hideUserErrorMessage: function () {
+                $('.error-message-banner').html('');
+            },
+
+            setElementVisibility: function (state) {
+                if (!this.model.isAboveMinimumAge()) {
+                    this.$('.upload-button-wrapper').css('display', state);
+                }
+
+                if (!this.model.has_profile_image()) {
+                    this.$('.u-field-remove-button').css('display', state);
+                }
+
+                if(this.inProgress() ||  this.options.editable === 'never') {
+                    this.$('.upload-button-wrapper').css('display', state);
+                    this.$('.u-field-remove-button').css('display', state);
+                }
+            },
+
+            addWindowActions: function () {
+                $(window).on('beforeunload', this.onBeforeUnload);
+            },
+
+            inProgress: function() {
+                var status = this.getCurrentStatus();
+                return _.isUndefined(status) ? false : true;
+            },
+
+            onBeforeUnload: function () {
+                var status = this.getCurrentStatus();
+                var message = gettext("{status} is in progress. Navigating away will abort it.");
```

Comment:
Will there only be one message whether the image is being uploaded or removed? If so, let's use this:

{status} is in progress. To avoid errors, stay on this page until the process is complete.